Add network support for RPI

DaniilKl opened the pull request:
Should be merged after https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/pull/1.

Commit:
env/: Implement Docker environment

Signed-off-by: Eduard Kaverinskyi <eduard.kaverinskyi@3mdeb.com>
Co-authored-by: Tymoteusz Burak <tymoteusz.burak@3mdeb.com>

WiktorG351 committed:
Dockerfile: add necessary deps

WiktorG351 committed:
env/build.sh: added

WiktorG351 committed:
env/Dockerfile: reorder packages neatly

WiktorG351 committed:
deleted build script, readme update

WiktorG351 committed:
Dockerfile: add qemu riscv deps and add build script for rpi4-ws demo

WiktorG351 committed:
remove tarballs, change debian version, update path

WiktorG351 committed:
fix export

WiktorG351 committed:
add working toolchain to dockerfile

WiktorG351 committed:
env/README.md: remove unnecessary docker create instruction

WiktorG351 committed:
env/README.md: change ubuntu to debian

WiktorG351 committed:
env/README.md: change adressing (no first person)

tym2k1 committed:
env/Dockerfile: start interactive shell at /work/crosscon

tym2k1 committed:
env/Dockerfile: use aarch64-*-gcc v11.2

tym2k1 committed:
env/Dockerfile: add ncurses

To be able to configure the end OS with `make menuconfig`

tym2k1 committed:
env/*: copy the local repository instead of pulling a fresh one

DaniilKl committed:
env: README: fix building hypervisor artifacts

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

tym2k1 committed:
env/README.md: standarize mountpoint path reference

tym2k1 committed:
env/Dockerfile: fix path when running from repo rootdir

tym2k1 committed:
env/files/env.sh: remove

We should be setting the environment variables in Dockerfile

tym2k1 committed:
env/Dockerfile: add libteec2 package from trixie repository

tym2k1 committed:
env/Dockerfile: reduce image size by cleaning at the end

tym2k1 committed:
rpi4-ws/configs/rpi4-single-vTEE/config.c: enable ethernet in REE in crosscon hv

tym2k1 committed:
support/linux-aarch64.config: CONFIG_BCMGENET=m -> CONFIG_BCMGENET=y

tym2k1 committed:
CROSSCON-Hypervisor: use patched fork

tym2k1 committed:
env/README.md: add instructions how to flash SD from inside container

WiktorG351 committed:
go back to old submodule setup

WiktorG351 committed:
rpi4-ws/README.md: add internet-enabling command

DaniilKl committed:
Merge pull request #6 from 3mdeb/develop

Merge to develop

tym2k1 commented:
Should the #1 be closed then?

DaniilKl commented:
Hmm, I see I missed the one you linked. :thinking: I guess it is better to leave only networking changes here, the rest (container and script) should be finished here https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/pull/1.

DaniilKl commented:
But networking changes are based on the container changes, the https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/pull/1 PR should be merged first :thinking:, will turn this PR to draft until the container PR is merged and rename it.

m-iwanicki committed:
support/linux-aarch64.config: add missing CROSSCONHYP_SHMEM config

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws/README.md: add '-f' to rm so it doesn't error & format text

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws: use SDCARD variable in scripts & cd to RPI4_WS dir before use

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws & security-test: add security test and minimal 2 VM config

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
Merge pull request #8 from 3mdeb/add-shared-mem-conf

support/linux-aarch64.config: add missing CROSSCONHYP_SHMEM config

dancesWithMachines committed:
rpi4-ws: Automate building demos

This commit aims to provide more automation for building CROSSCON
raspberry pi demos. Detailed changes:
* add non-root user to the container to avoid privileges rel. issues,
* simplify readme file and update commands,
* add scripts to build demos automatically.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Automate building demos pt.2

This commit adds additionall improvements to automation script for
building crosscon demos for RPI. Inlcuding:
* Added step selection
* Added configuration file selection
* Made the commands acutally re-runable (since step 1)
* Addressed minor picks

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Automate building demos pt.3

This commit add returning for ROOT dir at the beginning of every step
as futureproofing.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Automate building demos pt.4

This commit replaces "&& \" with "set -e".

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

macpijan committed:
env/build_rpi4.sh create missing optee dir

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
build_rpi4.sh: ingore error when creating bin dir

If we want it to trigger error, let's check for dir existence
excplictely and inform user what should be done in this case.

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
env/hyp_build_and_copy.sh: accept DEV_PATH from environment

Existing proposal is not only incovenient, but also dengerous,
possibly overwriting block device with no explicit confirmation.

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
env/hyp_build_and_copy.sh: add sync before umounting

umount does not sync to disk. It is a common problem that
data may get corrupted in this scenario.

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Build image

This commit gets rid of prepairing image via manual partition creation
and then copying files to SD card. Instead, a script that creates
ready-to-flash image has been prepaired.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Build image pt.2

This commit adds small improvements to script for creating demo image.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Build image pt.3

Remove redundant line for creating loopback device.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Build image pt.4

This patch changes image file creation so it is done as a non-root
user. This is to avoid issues with removing that file.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

pietrushnic committed:
Merge pull request #11 from 3mdeb/crscn-rpi4ws-build

rpi4-ws: Automate building demos

DaniilKl committed:
rpi4-ws: configs: rpi4-single-vTEE: delete interrupts 0 and 4 from Wifi
config

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
Merge pull request #7 from 3mdeb/network_support

Add network support

m-iwanicki committed:
CROSSCON-Hypervisor: use 3mdeb fork

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

DaniilKl committed:
CROSSCON-Hypervisor: add interrupt 27 fix

Fix for:

CROSSCONHYP INFO: VM 2 assigning interrupt 27
CROSSCONHYP ERROR: Trying to assign cpu interrupt multiple times

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
Merge pull request #12 from 3mdeb/fix-interrupt-27-hyp

CROSSCON-Hypervisor: add interrupt 27 fix

m-iwanicki commented on rpi4-ws/configs/rpi4-single-vTEE/config.c:
```diff
@@ -59,14 +59,22 @@ struct vm_config linux = {
+                .pa = 0x7d580000, // Physical address of the genet device
+                .va = 0x7d580000, // Virtual address for the VM
+                .size = 0x10000,  // Size of the device memory region
+                .interrupt_num = 4,
```
```suggestion
                .interrupt_num = 2,
```

DaniilKl committed:
rpi4-ws: configs: rpi4-single-vTEE: eth has 2 interrupts not 4

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
Merge pull request #13 from 3mdeb/correct-interrupts

Correct interrupts for wired network hardware

m-iwanicki committed:
rpi4-minimal.dts: explain why I'm not using whole assigned memory region

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
cache_test: fix comments and indentations

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
security_test: readme: add test commit

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
security_test: readme: explain arguments

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws: readme: fix config directory

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

DaniilKl committed:
env: build_rpi4.sh: add Cache Coloring security test as step 7

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
Merge pull request #14 from 3mdeb/security-test-daniil

Security test daniil

m-iwanicki committed:
security_test: add explicit cast

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

pietrushnic committed:
Merge pull request #10 from 3mdeb/security-test

Security test

m-iwanicki committed:
rpi4-ws: add minimal baremetal rpi4 app with UART

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
CROSSCON-Hypervisor: use commit with backported mem abort fix

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
mem separation: read and access address wanted by user

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws: add baremetal mem demo

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
.gitignore: add binary file to ignores

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
CROSSCON-Hypervisor: add printk to data abort function

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
memory-separation: include boot.S file

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws: README.md: add memory separation test section

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
mem_separation: use patch instead of commit

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
rpi4-ws: mem separation: update README.md

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

DaniilKl committed:
Merge pull request #18 from 3mdeb/mem-separation

Add demo for memory separation testing.

dancesWithMachines committed:
build_rpi4: Fix steps

This commit fixes env/build_rpi4.sh script so all steps are executed and
headers are properly printed.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
Merge pull request #19 from 3mdeb/build_rpi4-fix

build_rpi4: Fix steps

dancesWithMachines committed:
buildroot: make buildroot a submodule

Make buildroot a submodule to track changes.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
create_hyp_img: allow passing config

Allow specifying hypervior configs in create_hyp_img.sh.

Fixes:
https://github.com/3mdeb/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/issues/17

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
build_rpi4: Allow specifying device tree

Add option in build_rpi4 to allow using different DTSes.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
build_rpi4: Avoid unsets

Avoid unseting ENV variables, so other steps are unaffected.
Resotre CFLAGS in step 7 after compilation.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Integrate building scripts

Ditch all the scripts from rpi4-ws and integrate them with
env/build_rpi4.sh and env/create_hyp_image.sh. This is to unify building
across the demos.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: README: Update

Update README in rpi4-ws. All demos can now be build with single
env/build_rpi4.sh. The script will be transferred from ./env to
./rpi4-ws in incoming commits.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: Move stuff from env

Move rpi4-ws demo set related things from ./env to ./rpi4-ws and update
paths. Rename building script to just build.sh, update references in
README.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: build.sh: Allow passing args

Allow passing args so stuff can be run inside the container. Split the
docker command into multiple lines.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

DaniilKl committed:
Merge pull request #20 from 3mdeb/prep-upstream

Prepare for upstream

DaniilKl committed:
add CBA submodules

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
rpi4-ws: build.sh: add CBA stack building

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
cba: adhere to commits according to the context-based-auth-crosscon-demo
repo

The github.com/crosscon/context-based-auth-crosscon-demo uses these
commits on commit 58233e39ffebe63b14dd7403527b174b3b6f7ff1 which was
confirmed as working.

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
support: br-aarch64-cba.config: add buildroot config for CBA

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
rpi4-ws: rpi4-host-linux.dts: add

Copied from here:
https://github.com/crosscon/context-based-auth-crosscon-demo/blob/58233e39ffebe63b14dd7403527b174b3b6f7ff1/rpi4-ws/rpi4-host-linux.dts

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
nexmon: add nexmon VM artifacts

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
rpi4-ws: add dual GP dual TEE configuration

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: some small fixes

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection part 1, exclude two VM's: Nexmon and PKCS#11 VMs

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection part 2: delete the CBA OPTEE OS

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection: add the CBA OPTEE OS VM back

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection: add the Nexmon VM back

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection: fix hypervisor data aborts

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: bisection: fix CBA OPTEE OS shmem addresses

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
WIP: rpi4-ws: add PKCS#11 OPTEE OS VM

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
rpi4-ws: rpi4-per-vm-vTEE: fix second OPTEE OS probing issues

Previous logs:

[    3.382508] optee2: probing for conduit method.
[    3.387214] optee2: revision2 18446744073709551615.0
[    3.387226] optee2: api revision mismatch
[    3.396407] optee2: probe of soc:firmware:optee2 failed with error -22

With this commit:

[    3.380039] optee2: probing for conduit method.
[    3.384785] optee2: revision2 3.20 (d015c528)
[    3.384826] OPTEE2: Start 0xc8200000, size 0x200000
[    3.395115] optee2: initialized driver

It seems that the OPTEE OS VMs should be assosiated with the Linux VM in
some way in the CROSSCON Hypervisor configuration. Either as a parent or
as a child.

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
cba_ta: update to the newest development branch

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

DaniilKl committed:
cba_ta: add fix for issue #6

https://github.com/crosscon/context-based-auth-remote/issues/6

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

dancesWithMachines committed:
Add script for building RISC-V solution

Add build_riscv.sh script for building the solution for RISC-V arch.
in a docker container. Additionally update dockerfile to include
toolset for RISC-V baremetal compilation.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
nexmon: add nexmon.bin

Add nexmon binary and add an exception in .gitignore.
nexmon.bin is a binary supplied by crosscon.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
rpi4-ws: build.sh: add --local-confs support

This commit adds support for --local-confs parameter for
rpi4-ws/build.sh script. This param prevents overwritting local
configurations and just use .config files.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

DaniilKl committed:
Merge pull request #24 from 3mdeb/add_riscv_script

Add script for building RISC-V solution

DaniilKl committed:
Merge pull request #23 from 3mdeb/cba-integration

CBA integration
